[TensorExpr] Use schema instead of op name in NNC lowerings.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* __->__ #65843
* #65554

Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

Commit:
[TensorExpr] Use schema instead of op name in NNC lowerings.

Fixes #64963.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

navahgar commented on torch/csrc/jit/tensorexpr/lowerings.h:
```diff
@@ -36,9 +36,13 @@ TORCH_API std::unordered_map<std::string, NNCLoweringFunction>&
-    getNNCLoweringRegistry()[name] = fn;
+struct RegisterNNCLoweringsFunction {
+  RegisterNNCLoweringsFunction(
+      const std::vector<std::string>& names,
```
nit: s/names/schemas

navahgar reviewed (approved):
I trust that you have gotten all the schema variants for the ops we have. I didn't go through each one to check that. 

Overall LGTM

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31282334).

Commit:
Update on "[TensorExpr] Use schema instead of op name in NNC lowerings."


Fixes #64963.

Differential Revision: [D31282334](https://our.internmc.facebook.com/intern/diff/D31282334)

[ghstack-poisoned]
